### DISCUSSION TITLE
How does the server identifies label

### DISCUSSION THREAD

**Discussion by pippo73** [photoprism/photoprism#3645] [Q&A]
Hello all, I've just discovered this app, so I'm a total rookie.
One thing I was really astonished is the capability of the server to auto-label the photos.
How does it work? wich is the technology that allows it?
thanks
Leo

**Comment by graciousgrey** [comment]
Hi Leo,

You can find more information on the labels in our documentation and developer guide :)

- https://docs.photoprism.app/user-guide/organize/labels/
- https://docs.photoprism.app/developer-guide/metadata/classification/

**Reply by pippo73** [reply]
As far I understand it uses some kind of AI.
is this process local to my server, or it get in touch with some external service?
sorry but I'm new to this

**Reply by graciousgrey** [reply]
Image Classification happens completely on your server. 

See also the first question in our FAQ:

https://docs.photoprism.app/user-guide/faq/
